java/test/jmri/util/SystemNameComparatorTest.java

```diff
@@ -17,6 +17,32 @@ public void testCTor() {
         Assert.assertNotNull("exists", t);
     }
 
+    @Test
+    public void testNumericComparison() {
+        SystemNameComparator t = new SystemNameComparator();
+        
+        Assert.assertEquals("same IS100", 0, t.compare("IS100", "IS100"));
+
+        Assert.assertEquals("IS100 < IS101", -1, t.compare("IS100", "IS101"));
+        Assert.assertEquals("IS101 > IS100", +1, t.compare("IS101", "IS100"));
+
+        Assert.assertEquals("not same IS001 IS1", 2, t.compare("IS001", "IS1"));         // imperfect, but what it does
```

Comment:
Per [Comparator Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#compare-T-T-), `2` is not a valid value for a comparison result. This may cause problems with third-party APIs that check the result of a comparison is `1`.

Reply:
A further note: `Comparible.compareTo(...)` allows any non-zero value to be returned, but `Comparator.compare(o1, o2)` is only -1, 0, 1, so the last line of this method might need to be something like:
`return result > 0 ? 1 : result < 0 ? -1 : 0;`